README


This archive contains the distribution Math-Currency,
version 0.50:

  Exact Currency Math with Formatting and Rounding

This software is copyright (c) 2001 by John Peacock <jpeacock@cpan.org>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v6.005.



Changes
Revision history for Math::Currency

{{$NEXT}}

0.53  2023-07-06
    - Fix localize() so that it works correctly for perl 5.37.7 where
      currency_symbol is empty string

0.52  2017-07-29
    - make SEE ALSO pod entries clickable
    - don't include .travis.yml in the build
    - make signature test a dzil author test

0.51  2017-02-27
    - Add tests for legacy Math::BigInt bcmp() calling syntax
    - Update test for Ruble character changes in debian
    - Update dist repo metadata to reflect mschout as maintainer
    - Update POD to reflect mschout/github as maintainer and authoritative
      repository
    - Fix localeconv tests for FreeBSD 11

0.50  2016-07-03
    - do not try to run command "locale -a" on Windows
    - modernize code and clean up a bit
    - update new_currency script
    - updated locale definition for de_DE
    - switch to Dist::Zilla for releases, Module::Build no longer required.
    - Make $obj->bcmp($val) work like Math::BigFloat says it should [Thanks Daniel]
    - add en_US/USD locale modules

0.49  2016-06-21
    - fix russian currency symbol test comparisons
    - fix comparison tests for older Test::More
    - require Test::More 0.94 or later, require version.pm
    - UTF-8 fixes for localize()
    - avoid repetitive calls to "locale -a"
    - skip tests that require en_GB locale if that locale is not present
    - restore original locale at end of unknown_currency()

0.48  2016-06-16
    - fix compatibility with Math::BigInt >= v1.999717

2014-05-05  John Peacock  <john.peacock@havurah-software.org>

    * lib/Math/Currency.pm, lib/Math/Currency/de_DE.pm,
    lib/Math/Currency/en_GB.pm:
    Prep for new CPAN release
    [0351bc56d1da]

    * MANIFEST:
    Add missing META file
    [9070d564c73d]

    * MANIFEST.SKIP, lib/Math/Currency/de_DE.pm, scripts/new_currency:
    Fix up new_currency script and gen de_DE
    [fce0e05d0f67]

    * MANIFEST:
    Fix up MANIFEST
    [7e1b6bd2f564]

    * MANIFEST, t/002_basic.t:
    Fix stupid errors
    [74e0414be33c]

2010-08-25  John Peacock  <john.peacock@havurah-software.org>

    * lib/Math/Currency.pm, lib/Math/Currency/ja_JP.pm, t/002_basic.t,
    t/003_subclass.t, t/004_localize.t, t/004_localize.t.pl,
    t/006_defaults.t:
    Segregate all of the locale tests into their own file.
    [68bc293c233c]

2010-08-10  John Peacock  <john.peacock@havurah-software.org>

    * Build.PL, MANIFEST, lib/Math/Currency.pm, t/002_basic.t,
    t/003_subclass.t:
    Skip en_GB tests if that locale is not installed. Fix all other test
    failures.
    [98d03afd7d48]

2009-11-28  convert-repo  <convert-repo>

    * .hgtags:
    update tags
    [c0848782fbcf]

2008-05-08  jpeacock  <jpeacock>

    * MANIFEST, README, lib/Math/Currency.pm, lib/Math/Currency/GBP.pm,
    lib/Math/Currency/JPY.pm, lib/Math/Currency/en_GB.pm,
    lib/Math/Currency/ja_JP.pm, scripts/new_currency, t/003_subclass.t,
    t/004_localize.t:
    Overhaul the I18N locale support, in the hopes that it won't make me
    any crazier than it already has, and maybe pass all tests on other
    platforms.
    [bae8825fd549]

2008-01-31  jpeacock  <jpeacock>

    * lib/Math/Currency.pm, scripts/new_currency:
    Cope with more than one EUR format at once. Patch from JROBINSON
    <cpan @ desert-island.me.uk> plus some other tweaks by me.
    [fefc8d817f7e]

2007-09-23  jpeacock  <jpeacock>

    * Build.PL, lib/Math/Currency.pm, t/002_basic.t:
    as_int() could have floating point errors. See:

     http://rt.cpan.org/Public/Bug/Display.html?id=29210

    for details. This code is faster too! Patch by <rybskej @ yahoo.com>
    [7b38736783b1] [RELEASE_0.46]

2007-08-07  jpeacock  <jpeacock>

    * t/002_basic.t:
    Actually tests cleanly both with and without en_US
    [9688d279287b] [RELEASE_0.4502]

2007-08-06  jpeacock  <jpeacock>

    * lib/Math/Currency.pm, t/002_basic.t:
    Deal with problems where en_US isn't installed at all. Reported by
    David Cantrell <david@cantrell.org.uk>
    [be6d80cc22cd]

2007-08-05  jpeacock  <jpeacock>

    * lib/Math/Currency.pm, t/002_basic.t, t/003_subclass.t,
    t/005_rounding.t, t/006_defaults.t:
    Stupid locale() tests didn't work for non-US locale
    [72b6d6380712] [RELEASE_0.4501]

2007-08-04  jpeacock  <jpeacock>

    * MANIFEST:
    Forgot to add new testfile to the MANIFEST
    [9377aca59a3c] [RELEASE_0.45]

    * lib/Math/Currency.pm, t/006_defaults.t:
    Bugfix to correctly handle changing default currency as a class
    method. New tests for changing default currency as class method.
    [2abe0573fc75]

2006-01-25  jpeacock  <jpeacock>

    * MANIFEST, lib/Math/Currency.pm:
    Final changes before CPAN release.

    * MANIFEST Forgot to add new test here.

    * lib/Math/Currency.pm Test localeconv() early and bail immediately
    if not supported. Bad markup in POD.
    [5b5dc94eca04] [RELEASE_0_44]

2006-01-24  jpeacock  <jpeacock>

    * README, lib/Math/Currency.pm, lib/Math/Currency/GBP.pm,
    lib/Math/Currency/JPY.pm, t/002_basic.t, t/003_subclass.t,
    t/005_rounding.t:
    Prep for new release to CPAN.

    * README Note the changes for this release.

    * lib/Math/Currency.pm Add additional two digits of precision
    (hidden) to prevent loss of accuracy on multiple math operations.
    Suggested by Brian Phillips <brianp@holmescorp.com>

    * lib/Math/Currency/JPY.pm lib/Math/Currency/GBP.pm Run code through
    perltidy to try and make things consistent.

    * t/002_basic.t New empty subclass test.

    * t/003_subclass.t Loading class has to happen in BEGIN{} to be
    valid test.

    * t/005_rounding.t New tests for extended precision feature.
    [864b933c865a]

2005-12-16  jpeacock  <jpeacock>

    * README, lib/Math/Currency.pm, t/002_basic.t:
    * lib/Math/Currency.pm Preserve object formatting when making
    implicit copy, Reported by "Brian Phillips" <brianp@holmescorp.com>.
    Bump $VERSION.

    * t/002_basic.t Test for above.


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.005.
use strict;
use warnings;



use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Exact Currency Math with Formatting and Rounding",
  "AUTHOR" => "Michael Schout <mschout\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Math-Currency",
  "LICENSE" => "perl",
  "NAME" => "Math::Currency",
  "PREREQ_PM" => {
    "Encode" => 0,
    "Encode::Locale" => 0,
    "Exporter" => 0,
    "Math::BigFloat" => "1.60",
    "POSIX" => 0,
    "Test::More" => "0.94",
    "base" => 0,
    "overload" => 0,
    "strict" => 0,
    "utf8" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Socket" => 0,
    "Test::More" => "0.94",
    "Test::More::UTF8" => 0,
    "lib" => 0,
    "vars" => 0
  },
  "VERSION" => "0.50",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Encode" => 0,
  "Encode::Locale" => 0,
  "Exporter" => 0,
  "Math::BigFloat" => "1.60",
  "POSIX" => 0,
  "Socket" => 0,
  "Test::More" => "0.94",
  "Test::More::UTF8" => 0,
  "base" => 0,
  "lib" => 0,
  "overload" => 0,
  "strict" => 0,
  "utf8" => 0,
  "vars" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
